lib/elastomer/client/delete_by_query.rb

```diff
@@ -48,6 +48,27 @@ def delete_by_query(query, params = {})
       DeleteByQuery.new(self, query, params).execute
     end
 
+    SEARCH_PARAMETERS = [
+      :q,
+      :df,
+      :analyzer,
+      :analyze_wildcard,
+      :batched_reduce_size,
+      :default_operator,
+      :lenient,
+      :explain,
+      :_source,
+      :stored_fields,
+      :sort,
+      :track_scores,
+      :timeout,
+      :terminate_after,
+      :from,
+      :size,
+      :search_type,
+      :scroll
+    ].to_set.freeze
+
```

Comment:
I took all the valid search parameters from the doc. I'm not entirely sure we need to add all of them though.